xray/client_test.go

```diff
@@ -177,6 +178,62 @@ func TestRoundTripWithQueryParameter(t *testing.T) {
 	assert.Equal(t, headers.RootTraceID, seg.TraceID)
 }
 
+func TestRoundTripWithBasicAuth(t *testing.T) {
+	ctx, td := NewTestDaemon()
+	defer td.Close()
+
+	const content = `200 - Nothing to see`
+	const responseContentLength = len(content)
+
+	var userInfo = url.UserPassword("user", "pass")
+
+	ch := make(chan XRayHeaders, 1)
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		username, password, ok := r.BasicAuth()
+		pass, _ := userInfo.Password()
+		assert.Equal(t, ok, true)
+		assert.Equal(t, username, userInfo.Username())
+		assert.Equal(t, password, pass)
+		ch <- ParseHeadersForTest(r.Header)
+		w.WriteHeader(http.StatusOK)
+		if _, err := w.Write([]byte(content)); err != nil {
+			panic(err)
+		}
+	}))
+	defer ts.Close()
+
+	client := Client(nil)
+
+	u, err := url.Parse(ts.URL)
+	if !assert.NoError(t, err) {
+		return
+	}
+	u.User = userInfo
+
+	err = httpDoTest(ctx, client, http.MethodGet, u.String(), nil)
+	if !assert.NoError(t, err) {
+		return
+	}
+
+	seg, err := td.Recv()
+	if !assert.NoError(t, err) {
+		return
+	}
+	var subseg *Segment
+	if assert.NoError(t, json.Unmarshal(seg.Subsegments[0], &subseg)) {
+		assert.Equal(t, "remote", subseg.Namespace)
+		assert.Equal(t, http.MethodGet, subseg.HTTP.Request.Method)
+		assert.Equal(t, stripURL(*u), subseg.HTTP.Request.URL)
```

Comment:
We are testing the function against the function. We are not really testing if the functionality works. Let's test the output against a source of truth.

```suggestion
		assert.Equal(t, "http://user:***@127.0.0.1:"+u.Port(), subseg.HTTP.Request.URL)
```